### PR TITLE
Serde from

### PR DESCRIPTION
**Pull request by Licenser**

### CODE REVIEW

File: src/serde.rs [hunk 1]
```diff
@@ -7,12 +7,37 @@
...
+pub use self::value::*;
 use crate::numberparse::Number;
 use crate::{stry, Deserializer, Error, ErrorType, Result};
+use crate::{BorrowedValue, OwnedValue};
 use serde_ext::Deserialize;
+use std::convert::{TryFrom, TryInto};
 use std::fmt;
 
-pub use self::value::*;
+type ConvertResult<T> = std::result::Result<T, SerdeConversionError>;
```
Comment (sunnygleason) [inline src/serde.rs]: this makes sense 👍 

is it feasible to add test(s) to make sure patterns are covered?

File: src/serde.rs [hunk 2]
```diff
@@ -117,3 +142,135 @@ impl<'de> Deserializer<'de> {
         }
     }
 }
+
+impl TryFrom<serde_json::Value> for OwnedValue {
```
Comment (sunnygleason) [inline src/serde.rs]: 👍

File: src/serde.rs [hunk 3]
```diff
@@ -117,3 +142,135 @@ impl<'de> Deserializer<'de> {
...
+            Value::Object(o) => BorrowedValue::Object(
+                o.into_iter()
+                    .map(|(k, v)| Ok((k.into(), v.try_into()?)))
+                    .collect::<ConvertResult<crate::value::borrowed::Map>>()?,
+            ),
+        })
+    }
+}
+
+impl<'value> TryInto<serde_json::Value> for BorrowedValue<'value> {
```
Comment (sunnygleason) [inline src/serde.rs]: :+1:

**Comment by sunnygleason** [comment]
@Licenser this looks nice, thank you for putting it together!

File: src/serde.rs (hunk 1 above)
Comment (Licenser) [reply src/serde.rs]: Absolutely! It's not done yet going to add serde_yaml too, sorry I should have marked it as a draft.

**Comment by sunnygleason** [comment]
@Licenser I've created a separate PR for rustfmt/CI so it's a cleaner commit history, would you be able to rebase/cherry-pick?

**Comment by sunnygleason** [comment]
merged that other PR #44 w/ the whitespace & CI changes; sorry for jumping
in front of this PR, hopefully it makes it cleaner/easier to understand
at least!

**Comment by sunnygleason** [comment]
very nice!

File: src/serde.rs [hunk 4]
```diff
@@ -117,3 +142,204 @@ impl<'de> Deserializer<'de> {
...
+#[cfg(test)]
+mod test {
+    use crate::{json, BorrowedValue, OwnedValue};
+    use serde_json::{json as sjson, Value as SerdeValue};
+    use std::convert::TryInto;
+    #[test]
+    fn convert_owned_value() {
+        let v: OwnedValue = json!({
+            "int": 42,
+            "float": 7,
```
Comment (sunnygleason) [inline src/serde.rs]: curious, does a real float w/ decimal point not work because of round-trip inaccuracy?

Comment (Licenser) [reply src/serde.rs]: Nope that was just me being stupid :)

Comment (Licenser) [reply src/serde.rs]: good catch! 👍 fixed!

**Pull request merged**